### PR TITLE
Event handler

### PR DESCRIPTION
**Pull request by ChaabaneAnas**
## CHANGES

- add a click event handler function
- add state to the parent component (App) and reset it to null when `componentDidMount`
- lift the handler to the parent component to synchronize the child components
- transfer data to display components through props

### CODE REVIEW

File: src/Components/calculator.js [hunk 1]
```diff
@@ -1,19 +1,18 @@
-/* eslint-disable react/destructuring-assignment */
+/* eslint-disable jsx-a11y/click-events-have-key-events */
+/* eslint-disable jsx-a11y/no-static-element-interactions */
 /* eslint-disable react/prop-types */
+/* eslint-disable react/destructuring-assignment */
 /* eslint-disable react/prefer-stateless-function */
 /* eslint-disable max-classes-per-file */
 import React from 'react';
```
Comment (Munsa1) [inline src/Components/calculator.js]: - [x] It is a requirement that you should import `calcutate.js` into the `calculator.js` component. Kindly do so.

image

**Review by Munsa1 — CHANGES_REQUESTED** [review]
Hi @ChaabaneAnas, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!
<FEEL FREE TO CHANGE THIS MESSAGE TO YOUR PERSONAL WORDS>

## Highlights
:heavy_check_mark: Descriptive Readme
:heavy_check_mark: Passing Linters
:heavy_check_mark: Correct Gitflow
## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/Components/calculator.js (hunk 1 above)
Comment (ChaabaneAnas) [reply src/Components/calculator.js]: @Munsa1 are u sure that this is required for it to function...
I mean the only place a function from `calcutate.js` gets called is in the App.js after lifting the state up ...
and the only thing I'm getting after doing so is an ESlint error that says's` calculate defined but never been used  ` which is the case

**Review by KDlamini — APPROVED** [review]
### Hi @ChaabaneAnas 🙋‍♀️,  

You have done a great job implementing all the project requirements. 🎊 🥳 
Your feature is complete! There is nothing else to say other than... it's time to merge it. :shipit: 
Congratulations!!!

# Status: Approved ✔️ 

giphyapproved
 
Cheers 🥂  and Happy coding!!! 👯 
Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me @KDlamini  in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**